lib/cloud_controller/diego/protocol/routing_info.rb

```diff
@@ -56,18 +24,52 @@ def routing_info
         rescue RoutingApi::UaaUnavailable
           raise CloudController::Errors::ApiError.new_from_details('UaaUnavailable')
         end
-        # rubocop:enable Metrics/CyclomaticComplexity
 
         private
 
-        def route_id_app_ports_map
-          process.route_mappings(reload: true).each_with_object({}) do |route_mapping, route_app_port_map|
-            route_app_port_map[route_mapping.route_guid] ||= []
-            route_app_port_map[route_mapping.route_guid].push(get_port_to_use(process, route_mapping))
+        def http_info
+          relevant_http_routes = process.routes.reject do |route|
+            route.internal? ||
+              (route.tcp? &&
+                route.domain.router_group.present?)
```

Comment:
correct me if I am wrong, but routes can only be `tcp` if there is a router_group present, right?  I **think** the `route.domain.router_group.present?` check is superfluous

https://github.com/cloudfoundry/cloud_controller_ng/blob/main/app/models/runtime/shared_domain.rb#L74